rules/TypeDeclaration/TypeAnalyzer/GenericClassStringTypeNormalizer.php

```diff
@@ -69,44 +54,4 @@ private function resolveStringType(string $value): GenericClassStringType | Stri
         }
         return new StringType();
     }
-
-    private function verifyAutoImportedFullyQualifiedType(Type $type, bool $isAutoImport): ?Type
-    {
-        if ($type instanceof UnionType) {
-            $unionTypes = $type->getTypes();
-            $types = [];
-            $hasFullyQualifiedObjectType = false;
-            foreach ($unionTypes as $unionType) {
-                if ($this->isAutoImportFullyQualifiedObjectType($unionType, $isAutoImport)) {
-                    /** @var FullyQualifiedObjectType $unionType */
-                    $types[] = new GenericClassStringType(new ObjectType($unionType->getClassName()));
-                    $hasFullyQualifiedObjectType = true;
-                    continue;
-                }
-
-                $types[] = $unionType;
-            }
-
-            if ($hasFullyQualifiedObjectType) {
-                return new UnionType($types);
-            }
-
-            return $type;
-        }
-
-        if ($this->isAutoImportFullyQualifiedObjectType($type, $isAutoImport)) {
-            /** @var FullyQualifiedObjectType $type */
-            return new GenericClassStringType(new ObjectType($type->getClassName()));
-        }
-
-        return null;
-    }
-
-    private function isAutoImportFullyQualifiedObjectType(Type $type, bool $isAutoImport): bool
-    {
-        return $isAutoImport && $type instanceof FullyQualifiedObjectType && ! str_contains(
-            $type->getClassName(),
-            '\\'
-        );
-    }
```

Comment:
I'm not sure why thas was here... could you check the git blame on this? What PR added it?

Reply:
It is from PR https://github.com/rectorphp/rector-src/pull/457 . The tests is still in `tests/Issues/IssueImportedReturn ` directory for it.

The above code seems no longer needed after one of your PR, possibly  https://github.com/rectorphp/rector-src/pull/480

Reply:
I see, thanks for exploring. Mainly for the test case that still works 🙂